libminifi/test/flow-tests/FlowControllerTests.cpp

```diff
@@ -215,20 +220,19 @@ TEST_CASE("Extend the waiting period during shutdown", "[TestFlow4]") {
   testController.startFlow();
 
   // wait for the source processor to enqueue its flowFiles
-  int tryCount = 0;
-  while (tryCount++ < 10 && root->getTotalFlowFileCount() != 3) {
-    std::this_thread::sleep_for(std::chrono::milliseconds{20});
-  }
+  busy_wait(std::chrono::milliseconds{50}, [&] {return root->getTotalFlowFileCount() != 0;});
 
-  REQUIRE(root->getTotalFlowFileCount() == 3);
+  REQUIRE(root->getTotalFlowFileCount() != 0);
   REQUIRE(sourceProc->trigger_count.load() == 1);
 
   std::thread shutdownThread([&]{
     execSinkPromise.set_value();
     controller->stop(true);
   });
 
-  while (controller->isRunning()) {
+  int extendCount = 0;
+  while (extendCount++ < 5 && controller->isRunning()) {
```

Comment:
Maybe we could have more attempts with smaller timeout increases.

Reply:
shorter sleeps and checking for a timeout instead of run count should achieve the same goal without multiplying the uncertainty of the agents.

Reply:
note that here we are not simply waiting for the controller to shutdown, but we are verifying if we can progressively delay the shutdown by adjusting the timeout parameter, shorter sleeps seem to be relatively more imprecise than longer ones on the agents, I understand now that this loop might not be the best so I will unroll it to properly convey its purpose